### DISCUSSION TITLE
Improve startup time

### DISCUSSION THREAD

**Discussion by kico-itexto** [apache/grails-core#13542] [Ideas]
I was away from Grails for a while and recently started a new project using Grails 6.2: the startup time for the project is something that must be taken as a serious issue.

This happens in two moments: the first moment when you start a new project it's noticeable the amount of time is taken download dependencies (including Gradle). I remember that when Grails 1 whas still a thing this was not an issue.

This may sound silly, but as a first contact it may drive away people which could be interest on Grails. It's normal when we are using a new dependency, but when just creating a new project it's not a good first impression.

The second moment is on development, specially when using plugins like Spring Security. In development mode it still takes too long to start the project. In many projects I see people suffering with the time is taken to startup the system when running integration tests for example.

This is a rally old issue (the second moment), I don't know if there's something we could do to improve it, but still is an issue.

**Comment by gsartori** [comment]
I agree on the Spring Security Core issue, it would be good to improve its startup time.

**Reply by rainboyan** [reply]
Migrate to Spring Boot’s AutoConfiguration is the best way, the first reason is easily updated to latest Spring Security Framework, and the second is better performance, the third is that it is easier to support customization.
